src/traces/violin/defaults.js

```diff
@@ -26,9 +26,15 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     if(traceOut.visible === false) return;
 
     coerce('bandwidth');
-    coerce('scalegroup', traceOut.name);
-    coerce('scalemode');
     coerce('side');
+    var width = coerce('width');
+    if(!width) {
+        coerce('scalegroup', traceOut.name);
+        coerce('scalemode');
+    } else {
+        traceOut.scalegroup = '';
+        traceOut.scalemode = 'width';
+    }
```

Comment:
Does anything break if we 🔪 the `else` entirely? Unnecessary attributes should simply not be included in `traceOut` at all.

Reply:
yeah the violins are not drawn properly: the outlines are filled with rgb(0,0,0)